doc/ingresscontroller.md

```diff
@@ -247,7 +434,306 @@ Version: 2.0.0
+    api-addr: 0.0.0.0:2381
+    data-dir: /opt/eg-data/data
+    wal-dir: ""
+    cpu-profile-file: ""
+    memory-profile-file: ""
+    log-dir: /opt/eg-data/log
+    debug: false
+  controller.yaml: |
+    ...
+---
+```
+
+The URL `http://easegress-0.easegress-hs.default:2380` defined in `cluster.primary-listen-peer-urls` is an address of one cluster member, which *secondary* instance can use to connect to the cluster.
+
+Deployment could look like this:
+
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: easegress-ingress
+  name: easegress-secondary
+  namespace: default
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: easegress-ingress
+  template:
+    metadata:
+      labels:
+        app: easegress-ingress
+    spec:
+      serviceAccountName: easegress-ingress-controller
+      containers:
+      - args:
+        - -c
+        - |-
+          echo name: $EG_NAME > /opt/eg-config/config.yaml &&
+          cat /opt/eg-config/eg-secondary.yaml >> /opt/eg-config/config.yaml &&
+          /opt/easegress/bin/easegress-server \
+            -f /opt/eg-config/config.yaml \
+            --initial-object-config-files /opt/eg-config/controller.yaml
+        command:
+        - /bin/sh
+        env:
+        - name: EG_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        image: megaease/easegress:latest
+        imagePullPolicy: IfNotPresent
+        name: easegress-secondary
+        ports:
+        - containerPort: 2381
+          name: admin-port
+          protocol: TCP
+        - containerPort: 2380
+          name: peer-port
+          protocol: TCP
+        - containerPort: 2379
+          name: client-port
+          protocol: TCP
+        resources:
+          limits:
+            cpu: 1200m
+            memory: 2Gi
+          requests:
+            cpu: 100m
+            memory: 256Mi
+        volumeMounts:
+        - mountPath: /opt/eg-config/eg-secondary.yaml
+          name: easegress-cluster-cm
+          subPath: eg-secondary.yaml
+        - mountPath: /opt/eg-config/controller.yaml
+          name: easegress-cluster-cm
+          subPath: controller.yaml
+        - mountPath: /opt/eg-data/
+          name: ingress-data-volume
+      restartPolicy: Always
+      volumes:
+      - emptyDir: {}
+        name: ingress-data-volume
+      - configMap:
+          defaultMode: 420
+          items:
+          - key: eg-secondary.yaml
+            path: eg-secondary.yaml
+          - key: controller.yaml
+            path: controller.yaml
+          name: easegress-cluster-cm
+        name: easegress-cluster-cm
+---
+```
+
+Applying the deployment to the cluster creates two pods with prefix `easegress-secondary-`. We can again log into one of pods and see new *secondary* members.
+
+```bash
+# query admin port with egctl
+egctl --server {NODE_IP}:31255 member list |grep " name"
+# => prints
```

Comment:
propose to move the comments out of the shell script, some user may think they are commands to execute.